apps/omg_eth/test/omg_eth/ethereum_height_monitor_test.exs

```diff
@@ -0,0 +1,192 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Eth.EthereumHeightMonitorTest do
```

Comment:
this is cool testing!
there's only one suspicion, I've commented above as well. I'm wondering what will happen to your Monitor process if HTTP request perhaps returns and throws back a connection issue or ssl termination, is that handled on ethereumex? I can't remember rly.

Reply:
So I think it’s an error tuple all the way down from ethereumex -> httpoison -> hackney, at least from the typespecs and quick glance at the code.

- https://github.com/mana-ethereum/ethereumex/blob/master/lib/ethereumex/http_client.ex#L6-L19
- https://github.com/edgurgel/httpoison/blob/master/lib/httpoison/base.ex#L783-L860
- https://github.com/benoitc/hackney/blob/master/src/hackney.erl#L297-L389

So I think we're good here. I have some code for a new PR that could add this connection drop testing to `OMG.Eth`. As long as `OMG.Eth` doesn't raise an exception this monitor should be able to handle it.